extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultConfig.java

```diff
@@ -0,0 +1,227 @@
+/*
+ *  Copyright (c) 2022 Mercedes-Benz Tech Innovation GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Mercedes-Benz Tech Innovation GmbH - Initial API and Implementation
+ *       Mercedes-Benz Tech Innovation GmbH - Add token rotation mechanism
+ *
+ */
+
+package org.eclipse.edc.vault.hashicorp;
+
+import org.eclipse.edc.spi.EdcException;
+import org.eclipse.edc.spi.system.ServiceExtensionContext;
+
+import java.time.Duration;
+import java.util.Objects;
+
+import static java.lang.String.format;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_API_HEALTH_PATH;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_API_HEALTH_PATH_DEFAULT;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_API_SECRET_PATH;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_API_SECRET_PATH_DEFAULT;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_HEALTH_CHECK_ENABLED;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_HEALTH_CHECK_ENABLED_DEFAULT;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_HEALTH_CHECK_STANDBY_OK;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_HEALTH_CHECK_STANDBY_OK_DEFAULT;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_TIMEOUT_SECONDS;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_TIMEOUT_SECONDS_DEFAULT;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_TOKEN;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_TOKEN_RENEW_BUFFER;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_TOKEN_RENEW_BUFFER_SECONDS_DEFAULT;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_TOKEN_TTL;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_TOKEN_TTL_SECONDS_DEFAULT;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_URL;
+
+class HashicorpVaultConfig {
+
+    private String url;
+    private boolean healthCheckEnabled;
+    private String healthCheckPath;
+    private boolean healthStandbyOk;
+    private Duration timeout;
+    private String token;
+    private int timeToLive;
+    private int renewBuffer;
+    private String secretPath;
+
+    private HashicorpVaultConfig() {}
+
+    public static HashicorpVaultConfig create(ServiceExtensionContext context) {
```

Comment:
Please refactor this into a method that is used by the extension. Passing a context to a static that uses a builder like this is convoluted and pollutes it with extension concepts.